docs/source/format/Glossary.rst

```diff
@@ -0,0 +1,207 @@
+       :term:`slots <slot>` or singular values, even though for some
+       data types (like structs or unions), a slot may represent
+       multiple values.
+
+       Defined by the :doc:`./Columnar`.
+
+   buffer
+       A *contiguous* region of memory with a given length.  Buffers
+       are used to store data for arrays.
+
+       Buffers may be in CPU memory, memory-mapped from a file, in
+       device (e.g. GPU) memory, etc., though not all Arrow
+       implementations support all of these possibilities.
+
+   child array
+   parent array
+       In an array of a :term:`nested type`, the parent array
+       corresponds to the :term:`parent type` and the child array(s)
+       correspond to the :term:`child type(s) <child type>`.  For
+       example, a ``List[Int32]``-type parent array has an
+       ``Int32``-type child array.
+
+   child type
+   parent type
+       In a :term:`nested type`, the nested type is the parent type,
+       and the child type(s) are its parameters.  For example, in
+       ``List[Int32]``, ``List`` is the parent type and ``Int32`` is
+       the child type.
+
+   chunked array
+       A *discontiguous*, *one-dimensional* sequence of values with
+       known length where all values have the same type.  Consists of
+       zero or more :term:`arrays <array>`, the "chunks".
+
+       Chunked arrays are discontiguous in the sense that iterating
+       the values of a chunked array may require iterating through
+       different buffers for different indices.
+
+       Not part of the columnar format; this term is specific to
+       certain language implementations of Arrow (primarily C++ and
+       its bindings).
+
+       .. seealso:: :term:`record batch`, :term:`table`
+
+   complex type
+   nested type
+       A :term:`data type` whose structure depends on one or more
+       other :term:`child data types <child type>`. For instance,
+       ``List`` is a nested type that has one child.
+
+       Two nested types are equal if and only if their child types are
+       also equal.
+
+   data type
+   type
+       A type that a value can take, such as ``Int8`` or
+       ``List[Utf8]``. The type of an array determines how its values
+       are laid out in memory according to :doc:`./Columnar`.
+
+       .. seealso:: :term:`logical type`, :term:`nested type`,
+                    :term:`primitive type`
+
+   dictionary
+       An array of values that accompany a :term:`dictionary-encoded
+       <dictionary-encoding>` array.
+
+   dictionary-encoding
+       An array that stores its values as indices into a
+       :term:`dictionary` array instead of storing the values
+       directly.
+
+       .. seealso:: :ref:`dictionary-encoded-layout`
+
+   extension type
+       A user-defined :term:`data type` that adds additional semantics
+       to an existing data type.  This allows implementations that do
+       not support a particular extension type to still handle the
+       underlying data type.
```

Comment:
Do we want to use the term "storage type" for this underlying type?

Reply:
Ah, good catch. I also added "storage type" as an alias for this entry (I don't think it deserved its own entry).